File: Universe_Backend/Data/Models/StoryReaction.cs
using Universe_Backend.Data.DTOs;

namespace Universe_Backend.Data.Models;

public class StoryReaction : Reaction
{
    public int StoryId { get; set; }
    public virtual Story? Story { get; set; }

    public StoryReactionDTO ToDTO()
    {
        return new StoryReactionDTO
        {
            Id = Id,
            UserId = UserId,
            ReactionType = ReactionType,
            ReactionDate = ReactionDate,
            StoryId = StoryId
        };
    }
}

File: universe/lib/models/data/story_reaction.dart
import 'package:json_annotation/json_annotation.dart';

part 'story_reaction.g.dart';

@JsonSerializable()
class StoryReaction {
  int id;
  String userId;
  String reactionType;
  DateTime reactionDate;
  int storyId;

  StoryReaction({
    required this.id,
    required this.userId,
    required this.reactionType,
    required this.reactionDate,
    required this.storyId,
  });

  factory StoryReaction.fromJson(Map<String, dynamic> data) =>
      _$StoryReactionFromJson(data);

  Map<String, dynamic> toJson() => _$StoryReactionToJson(this);
}
